src/auth/auth.service.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { UsersService } from 'src/users/users.service';
+import * as bcrypt from 'bcrypt';
+//import { JwtService } from '@nestjs/jwt';
+
+@Injectable()
+export class AuthService {
+  constructor(private usersService: UsersService){}
+
+  async validateUser(username: string, pass: string): Promise<any> {
+    const user = await this.usersService.findOne(username);
+    const isMatch = await bcrypt.compare(pass, user.password);    // Authenticating Password.
+    if (user && isMatch) {
```

Comment:
Validate data in DTO.

Reply:
Here we are checking User availability and validity both from the DB. And the Documentation was suggested to keep this method in 'auth.service' file. 
Ref: https://docs.nestjs.com/security/authentication